README.md

```diff
@@ -1,123 +1,343 @@
 ```
-export QUERY="select ID,Card_Number,Card_Holders_Name from \`${PROJECT_ID}.${BQ_DATASET_NAME}.100000CCRecords\` where safe_cast(Credit_Limit as int64)>100000 and safe_cast (Age as int64)>50 group by ID,Card_Number,Card_Holders_Name limit 10"
+gradle build ... -Pargs="... --columnDelimiter=|"
 ```
-Create a gcs file with the query:
+
+
+
+### S3 Scanner
+
+To use AWS S3 as a source of input files, use AWS credentials as instructed below.
+
+Export the AWS access key, secret key, and credentials provider to environment variables. 
 
 ```
-export GCS_REID_QUERY_BUCKET=<name>
-cat << EOF | gsutil cp - gs://${REID_QUERY_BUCKET}/reid_query.sql
-${QUERY}
-EOF
+export AWS_ACCESS_KEY="<access_key>"
+export AWS_SECRET_KEY="<secret_key>"
+export AWS_CRED="{\"@type\":\"AWSStaticCredentialsProvider\",\"awsAccessKeyId\":\"${AWS_ACCESS_KEY}\",\"awsSecretKey\":\"${AWS_SECRET_KEY}\"}"
 ```
-Run the pipeline by passing required parameters:
+
+Use Gradle to build and run the job to performs data loss prevention (DLP) on a CSV file stored in Amazon S3. The results will be written to BigQuery.
+
 ```
-gradle run -DmainClass=com.google.swarm.tokenization.DLPTextToBigQueryStreamingV2 -Pargs="--region=<region> --project=<project_id> --streaming --enableStreamingEngine --tempLocation=gs://<bucket>/temp --numWorkers=5 --maxNumWorkers=10 --runner=DataflowRunner --tableRef=<project_id>:<dataset>.<table> --dataset=<dataset> --topic=projects/<project_id>/topics/<name> --autoscalingAlgorithm=THROUGHPUT_BASED --workerMachineType=n1-highmem-4 --deidentifyTemplateName=projects/<project_id>/deidentifyTemplates/<name> --DLPMethod=REID --keyRange=1024 --queryPath=gs://<gcs_reid_query_bucket>/reid_query.sql"
+gradle spotlessApply
+
+gradle build
 
+// inspect is default as DLP Method; For deid: --DLPMethod=DEID
+gradle run -DmainClass=com.google.swarm.tokenization.DLPTextToBigQueryStreamingV2 -Pargs="--region=<region> --project=<project_id> --streaming --enableStreamingEngine --tempLocation=gs://<bucket>/temp --numWorkers=1 --maxNumWorkers=2 --runner=DataflowRunner --filePattern=s3://<bucket>>/file.csv --dataset=<name>  --inspectTemplateName=<inspect_template> --deidentifyTemplateName=<deid_tmplate> --awsRegion=<aws_region> --awsCredentialsProvider=$AWS_CRED"
 ```
+
+#### Parameters:
+
+* --awsRegion: The region where the AWS resources reside.
+
+* --awsCredentialsProvider: The AWS credentials provider.
+
+
+## How to adapt this pipeline for your use cases
+
+1. [Create your own inspection templates & run inspection on sample data](https://cloud.google.com/dlp/docs/creating-templates-inspect)
+2. [Create de-identification templates and run de-identification on sample data](https://cloud.google.com/dlp/docs/creating-templates-deid)
+3. In case of cross-regional data source or sink, dataflow service account should be granted necessary permissions.
+
+
+## Troubleshooting
```

Comment:
The explanations and recommendations are being rendered as blockquotes (like those you might see in email or chat threads). Consider making them regular paragraphs rather than blockquote elements.